File: for-ais-only/examples/go-redis/sample_test.go
// =============================================================================
// CANONICAL GO-REDIS TEST FILE TEMPLATE
// =============================================================================
// This file demonstrates the structure and conventions used for go-redis
// documentation test files. These tests serve dual purposes:
// 1. Executable tests that validate code snippets
// 2. Source for documentation code examples (processed via special markers)
//
// MARKER REFERENCE:
// - EXAMPLE: <name>     - Identifies the example name (matches docs folder name)
// - BINDER_ID <id>      - Optional identifier for online code runners
// - HIDE_START/HIDE_END - Code hidden from documentation but executed in tests
// - REMOVE_START/REMOVE_END - Code removed entirely from documentation output
// - STEP_START <name>/STEP_END - Named code section for targeted doc inclusion
//
// Go uses "Testable Examples" with // Output: comments for verification
// RUN: go test -v sample_test.go
// =============================================================================

// EXAMPLE: sample_example
// HIDE_START
package example_commands_test

import (
	"context"
	"fmt"

	"github.com/redis/go-redis/v9"
)

// HIDE_END

func ExampleClient_string_ops() {
	ctx := context.Background()

	rdb := redis.NewClient(&redis.Options{
		Addr:     "localhost:6379",
		Password: "",
		DB:       0,
	})

	// REMOVE_START
	rdb.Del(ctx, "mykey")
	// REMOVE_END

	// STEP_START string_ops
	res1, err := rdb.Set(ctx, "mykey", "Hello", 0).Result()
	if err != nil {
		panic(err)
	}
	fmt.Println(res1) // >>> OK

	res2, err := rdb.Get(ctx, "mykey").Result()
	if err != nil {
		panic(err)
	}
	fmt.Println(res2) // >>> Hello
	// STEP_END

	// Output:
	// OK
	// Hello
}

func ExampleClient_hash_ops() {
	ctx := context.Background()

	rdb := redis.NewClient(&redis.Options{
		Addr:     "localhost:6379",
		Password: "",
		DB:       0,
	})

	// REMOVE_START
	rdb.Del(ctx, "myhash")
	// REMOVE_END

	// STEP_START hash_ops
	res1, err := rdb.HSet(ctx, "myhash", "field1", "value1").Result()
	if err != nil {
		panic(err)
	}
	fmt.Println(res1) // >>> 1

	res2, err := rdb.HSet(ctx, "myhash",
		"field2", "value2",
		"field3", "value3",
	).Result()
	if err != nil {
		panic(err)
	}
	fmt.Println(res2) // >>> 2

	res3, err := rdb.HGet(ctx, "myhash", "field1").Result()
	if err != nil {
		panic(err)
	}
	fmt.Println(res3) // >>> value1

	res4, err := rdb.HGetAll(ctx, "myhash").Result()
	if err != nil {
		panic(err)
	}
	fmt.Println(res4["field1"]) // >>> value1
	// STEP_END

	// Output:
	// 1
	// 2
	// value1
	// value1
}

func ExampleClient_hash_tutorial() {
	ctx := context.Background()

	rdb := redis.NewClient(&redis.Options{
		Addr:     "localhost:6379",
		Password: "",
		DB:       0,
	})

	// REMOVE_START
	rdb.Del(ctx, "bike:1", "bike:1:stats")
	// REMOVE_END

	// STEP_START hash_tutorial
	bike1 := map[string]interface{}{
		"model": "Deimos",
		"brand": "Ergonom",
		"type":  "Enduro bikes",
		"price": 4972,
	}

	res1, err := rdb.HSet(ctx, "bike:1", bike1).Result()
	if err != nil {
		panic(err)
	}
	fmt.Println(res1) // >>> 4

	res2, err := rdb.HGet(ctx, "bike:1", "model").Result()
	if err != nil {
		panic(err)
	}
	fmt.Println(res2) // >>> Deimos

	res3, err := rdb.HGet(ctx, "bike:1", "price").Result()
	if err != nil {
		panic(err)
	}
	fmt.Println(res3) // >>> 4972
	// STEP_END

	// Output:
	// 4
	// Deimos
	// 4972
}

File: .agents/skills/generate-tce-examples/assets/rust-sync/sample_test.rs
// =============================================================================
// CANONICAL RUST-SYNC (redis-rs) TEST FILE TEMPLATE
// =============================================================================
// This file demonstrates the structure and conventions used for redis-rs sync
// documentation test files. These tests serve dual purposes:
// 1. Executable tests that validate code snippets
// 2. Source for documentation code examples (processed via special markers)
//
// MARKER REFERENCE:
// - EXAMPLE: <name>     - Identifies the example name (matches docs folder name)
// - BINDER_ID <id>      - Optional identifier for online code runners
// - HIDE_START/HIDE_END - Code hidden from documentation but executed in tests
// - REMOVE_START/REMOVE_END - Code removed entirely from documentation output
// - STEP_START <name>/STEP_END - Named code section for targeted doc inclusion
//
// RUN: cargo test
// =============================================================================

// EXAMPLE: sample_example
#[cfg(test)]
mod sample_tests {
    use redis::Commands;
    use std::collections::HashMap;

    #[test]
    fn run() {
        let mut r = match redis::Client::open("redis://127.0.0.1") {
            Ok(client) => match client.get_connection() {
                Ok(conn) => conn,
                Err(e) => {
                    println!("Failed to connect to Redis: {e}");
                    return;
                }
            },
            Err(e) => {
                println!("Failed to create Redis client: {e}");
                return;
            }
        };

        // REMOVE_START
        // Clean up any existing data before tests
        let _: Result<i32, _> = r.del(&["mykey", "myhash", "bike:1", "bike:1:stats"]);
        // REMOVE_END

        // STEP_START string_ops
        // Basic string SET/GET operations
        match r.set("mykey", "Hello") {
            Ok(res1) => {
                let res1: String = res1;
                println!("{res1}"); // >>> OK
                // REMOVE_START
                assert_eq!(res1, "OK");
                // REMOVE_END
            }
            Err(e) => println!("Error: {e}"),
        }

        match r.get("mykey") {
            Ok(res2) => {
                let res2: String = res2;
                println!("{res2}"); // >>> Hello
                // REMOVE_START
                assert_eq!(res2, "Hello");
                // REMOVE_END
            }
            Err(e) => println!("Error: {e}"),
        }
        // STEP_END

        // REMOVE_START
        let _: Result<i32, _> = r.del("mykey");
        // REMOVE_END

        // STEP_START hash_ops
        // Hash operations: HSET, HGET, HGETALL
        match r.hset("myhash", "field1", "value1") {
            Ok(res3) => {
                let res3: i32 = res3;
                println!("{res3}"); // >>> 1
                // REMOVE_START
                assert_eq!(res3, 1);
                // REMOVE_END
            }
            Err(e) => println!("Error: {e}"),
        }

        let hash_fields = vec![("field2", "value2"), ("field3", "value3")];
        match r.hset_multiple("myhash", &hash_fields) {
            Ok(res4) => {
                let res4: String = res4;
                println!("{res4}"); // >>> OK
                // REMOVE_START
                assert_eq!(res4, "OK");
                // REMOVE_END
            }
            Err(e) => println!("Error: {e}"),
        }

        match r.hget("myhash", "field1") {
            Ok(res5) => {
                let res5: String = res5;
                println!("{res5}"); // >>> value1
                // REMOVE_START
                assert_eq!(res5, "value1");
                // REMOVE_END
            }
            Err(e) => println!("Error: {e}"),
        }

        match r.hgetall("myhash") {
            Ok(res6) => {
                let res6: HashMap<String, String> = res6;
                println!("{:?}", res6.get("field1")); // >>> Some("value1")
                // REMOVE_START
                assert_eq!(res6.get("field1"), Some(&"value1".to_string()));
                // REMOVE_END
            }
            Err(e) => println!("Error: {e}"),
        }
        // STEP_END

        // REMOVE_START
        let _: Result<i32, _> = r.del("myhash");
        // REMOVE_END

        // STEP_START hash_tutorial
        // Tutorial-style example with bike data
        let bike_fields = vec![
            ("model", "Deimos"),
            ("brand", "Ergonom"),
            ("type", "Enduro bikes"),
            ("price", "4972"),
        ];

        match r.hset_multiple("bike:1", &bike_fields) {
            Ok(res7) => {
                let res7: String = res7;
                println!("{res7}"); // >>> OK
            }
            Err(e) => println!("Error: {e}"),
        }

        match r.hget("bike:1", "model") {
            Ok(res8) => {
                let res8: String = res8;
                println!("{res8}"); // >>> Deimos
                // REMOVE_START
                assert_eq!(res8, "Deimos");
                // REMOVE_END
            }
            Err(e) => println!("Error: {e}"),
        }

        match r.hget("bike:1", "price") {
            Ok(res9) => {
                let res9: String = res9;
                println!("{res9}"); // >>> 4972
                // REMOVE_START
                assert_eq!(res9, "4972");
                // REMOVE_END
            }
            Err(e) => println!("Error: {e}"),
        }
        // STEP_END

        // REMOVE_START
        let _: Result<i32, _> = r.del(&["bike:1", "bike:1:stats"]);
        // REMOVE_END
    }
}

